src/hooks/useList.js

```diff
@@ -52,11 +69,39 @@ export default function useList() {
     if (list.byId[targetId].isChecked) {
       //listItem moves up to unchecked items
       setUncheckedIds([...uncheckedIds, targetId])
-      setCheckedIds([...checkedIds.filter((id) => id !== targetId)])
+      setCheckedIds(checkedIds.filter((id) => id !== targetId))
     } else {
       //listItem moves down to checked items
-      setUncheckedIds([...uncheckedIds.filter((id) => id !== targetId)])
+      setUncheckedIds(uncheckedIds.filter((id) => id !== targetId))
       setCheckedIds([targetId, ...checkedIds])
     }
   }
+
+  function deleteListItem(targetId) {
+    const copyOfById = { ...list.byId }
+    delete copyOfById[targetId]
+    const byIdWithoutTargetId = copyOfById
+
```

Comment:
The naming is pretty hard to understand here ( maybe it`s just me )

Reply:
I don't like it either, but I couldn't come up with an easier solution. I even added the line `const byIdWithoutTargetId = copyOfById` in order to make it easier to understand. I also could have left that out and kept on working with `copyOfById` which doesn't include the deleted part anymore...
Do you have a better idea?